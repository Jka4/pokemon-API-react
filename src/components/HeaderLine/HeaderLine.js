import React from 'react';
import SearchInput from "../SearchInput/SearchInput";
import Player from "../Player/Player";
import './styles/style.scss';

const HeaderLine = () => {
  return (
    <div className='topLine'>
      <Player />
      <SearchInput />
    </div>
  );
}

export default HeaderLine;
