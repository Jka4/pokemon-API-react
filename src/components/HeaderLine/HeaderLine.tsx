import React from "react";
import { NavLink } from "react-router-dom";

import store from "../../Store/store";

import Search from "../Search/Search";
import Player from "../Player/Player";

import AppBar from "@material-ui/core/AppBar";
import { Button } from "@material-ui/core";
import SvgIcon from "@material-ui/core/SvgIcon";

import "./styles/style.scss";

interface HeaderLineProps {
  pathname?: string;
  isMainPage?: any;
}

const HeaderLine: React.FC<HeaderLineProps> = ({
  pathname,
}: HeaderLineProps) => {
  const isMainPage = pathname === "/";

  function HomeIcon({ props }: any) {
    return (
      <SvgIcon {...props}>
        <path d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z" />
      </SvgIcon>
    );
  }

  return (
    <>
      <AppBar position="static" className="topLine">
        <div className="buttonsRow">
          <Player />

          {!isMainPage && (
            <NavLink
              to="/"
              onClick={() => store.dispatch({ type: "CLEAR_DETAILS_PAGE" })}
            >
              <Button variant="contained" className="toHome" color="secondary">
                <HomeIcon />
              </Button>
            </NavLink>
          )}
        </div>
        <Search />
      </AppBar>
    </>
  );
};

export default HeaderLine;
