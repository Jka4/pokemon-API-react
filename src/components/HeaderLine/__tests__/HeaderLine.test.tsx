import * as React from 'react';
import ReactDOM from 'react-dom';
import { act } from 'react-dom/test-utils';
import HeaderLine from '../HeaderLine';

import { Provider } from 'react-redux';

import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { BrowserRouter } from 'react-router-dom';

describe('<HeaderLine/> ', () => {
  let container: any;

  beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    document.body.removeChild(container);
    container = null;
  });

  const mockStore = configureMockStore([thunk]);

  it('renders correctly', async () => {
    const store = mockStore({ pokemonArr: [] });

    await act(async () => {
      ReactDOM.render(
        <Provider store={store}>
          <BrowserRouter basename="/">
            <HeaderLine />
          </BrowserRouter>
        </Provider>,
        await container,
      );

      expect(container).toMatchSnapshot();
    });
  });
});
