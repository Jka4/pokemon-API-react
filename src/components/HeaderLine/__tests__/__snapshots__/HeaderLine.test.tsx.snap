// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderLine/>  renders with home button 1`] = `
<path
  d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
/>
`;

exports[`<HeaderLine/>  renders without home button 1`] = `null`;
